torch/testing/_internal/common_methods_invocations.py

```diff
@@ -19744,6 +19744,13 @@ def __init__(
                          dtypes=(torch.bfloat16,), device_type='cpu'),
         ),
     ),
+    ElementwiseBinaryPythonRefInfo(
+        "_refs.gcd",
+        torch_opinfo_name="gcd",
+        skips=(
+            DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_python_ref_errors'),
```

Comment:
This will need a comment -- why is this test failing?

Reply:
I not sure exactly why. Something related to meta tensor?
```
======================================================================
FAIL: test_python_ref_errors__refs_gcd_cpu (__main__.TestCommonCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/syeahmed/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 377, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/syeahmed/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 786, in test_wrapper
    return test(*args, **kwargs)
  File "/home/syeahmed/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 821, in dep_fn
    return fn(slf, *args, **kwargs)
  File "/home/syeahmed/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 979, in only_fn
    return fn(self, *args, **kwargs)
  File "/home/syeahmed/workspace/pytorch/test/test_ops.py", line 521, in test_python_ref_errors
    op(meta_sample.input, *meta_sample.args, **meta_sample.kwargs)
AssertionError: Exception not raised
```

Reply:
I don't think it's related to meta tensors (although the stack trace is confusing). Looks like some error is just not being thrown correctly. That's OK for now, though, we can take a look later. It's probably something very minor (like torch.gcd fails on scalars but refs.gcd accepts them).

Reply:
We should fix it now.

Reply:
OK... well to fix it now, @syed-ahmed ...

- Identify which error input is causing the failure, and print its details (this is a good opportunity to learn more about debugging the tests and handling sample inputs); you probably want to add a print statement to the `test_python_ref_errors` test that prints the error input, then print the relevant parts
- recreate that input in a Python script and pass it to torch.gcd and refs.gcd -- see what the difference is
- if the difference is scalar handling, set the Python reference OpInfo metadata for scalar handling properly (there are some other Python reference OpInfos that do this) 

ping me on Slack if you have any questions

Reply:
@mruberry As discussed in slack, the error case I'm failing on is when I have a meta object and scalar as an input. I marked the scalar handling in `_make_elementwise_binary_reference` instead of OpInfo. I tried setting:
```
supports_rhs_python_scalar=False
supports_one_python_scalar=False,
supports_two_python_scalars=False,
```
in OpInfo for gcd ref, but that still causes `test_python_ref_errors` to error out. Setting `supports_rhs_python_scalar` and `supports_lhs_python_scalar` in the make function raises the exception and hence, passes the test.

Without marking scalar handling, calls to `refs.gcd` with meta object silently executes and outputs nothing. With the handling, `refs.gcd` with meta object and scalars now raise an exception:
```
>>> import torch._refs as refs
>>> import torch._prims as prims
>>> import torch
>>> a_in = torch.tensor([1,2,3])
>>> a_in_meta = prims.utils.TensorMeta(a_in)
>>> refs.gcd(a_in, a_in)
tensor([1, 2, 3])
>>> refs.gcd(a_in_meta, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/home/pytorch/torch/_prims/wrappers.py", line 174, in _fn
    result = fn(*args, **kwargs)
  File "/opt/home/pytorch/torch/_prims/wrappers.py", line 115, in _fn
    result = fn(**bound.arguments)
  File "/opt/home/pytorch/torch/_refs/__init__.py", line 680, in _ref
    raise ValueError(
ValueError: Received a rhs Python scalar to an elementwise binary operation that does not accept rhs scalars!
>>> refs.gcd(a_in_meta, a_in_meta)

>>> refs.gcd(2, a_in_meta)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/home/pytorch/torch/_prims/wrappers.py", line 174, in _fn
    result = fn(*args, **kwargs)
  File "/opt/home/pytorch/torch/_prims/wrappers.py", line 115, in _fn
    result = fn(**bound.arguments)
  File "/opt/home/pytorch/torch/_refs/__init__.py", line 675, in _ref
    raise ValueError(
ValueError: Received a lhs Python scalar to an elementwise binary operation that does not accept lhs scalars!
>>> 
```